packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -1361,3 +1384,33 @@ export function isFavoriteChannel(state: GlobalState, channelId: string): boolea
 
     return category.channel_ids.includes(channel.id);
 }
+export function filterChannelList(channelList: Channel[], filters: ChannelSearchOpts): Channel[] {
+    if (!filters || (!filters.private && !filters.public && !filters.deleted && !filters.team_ids)) {
+        return channelList;
+    }
+    let result: Channel[] = [];
+    const channels = channelList;
+    if (filters.public) {
+        result = result.concat(channels.filter((channel) => channel.type === Constants.OPEN_CHANNEL));
+    }
+    if (filters.private) {
+        result = result.concat(channels.filter((channel) => channel.type === Constants.PRIVATE_CHANNEL));
+    }
+    if (filters.deleted) {
+        result = result.concat(channels.filter((channel) => channel.type === Constants.ARCHIVED_CHANNEL));
```

Comment:
for these channel types we can just do

let channelType

if (public)
channelType = Constants.OPEN_CHANNEL
....

result = result.concat( ..... channel.type === channelType)


Could cut down on code repitition

Reply:
Made some changes here. Take a look